lib/waistband/index.rb

```diff
@@ -314,6 +349,24 @@ def base_config_name
         "#{@index_name}_#{::Waistband.config.env}"
       end
 
+      def check_permission!(permission)
+        raise "::Waistband::Errors::Permissions::#{permission.classify}".constantize.new("Don't have enough permissions to #{permission}") unless check_permission?(permission)
+      end
+
+      def check_permission?(permission)
+        !!permissions[permission]
+      end
+
+      def permissions
+        @permissions ||= (config['permissions'] || {}).reverse_merge({
```

Comment:
The defaults are not false.   If not defined everything's allowed.
On Feb 22, 2015 11:26 AM, "Evan Tahler" <notifications@github.com> wrote:

> In lib/waistband/index.rb
> <https://github.com/taskrabbit/waistband/pull/12#discussion_r25135460>:
>
> > @@ -314,6 +349,24 @@ def base_config_name
> >          "#{@index_name}_#{::Waistband.config.env}"
> >        end
> >
> > +      def check_permission!(permission)
> > +        raise "::Waistband::Errors::Permissions::#{permission.classify}".constantize.new("Don't have enough permissions to #{permission}") unless check_permission?(permission)
> > +      end
> > +
> > +      def check_permission?(permission)
> > +        !!permissions[permission]
> > +      end
> > +
> > +      def permissions
> > +        @permissions ||= (config['permissions'] || {}).reverse_merge({
>
> Did you pick these defaults to be backwards-compatibile? I would suggest
> more safety here for the defaults (delete = false, destroy = false).
> Perhaps default_permissions should be its own method, to it can be
> overwritten?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskrabbit/waistband/pull/12/files#r25135460>.
>